src/utils/avatar.js

```diff
@@ -391,7 +391,7 @@ export class UploadedAvatarURL extends AvatarURL {
 
       const match = /\.png$/.exec(result.pathname);
       if (match !== null) {
-        result.pathname = result.pathname.replace(match[0], '-medium.png');
+        result.pathname = result.pathname.replace(/\.png$/, '-medium.png');
```

Comment:
As the commit message suggests, this potentially does the wrong thing before this commit. I believe none of the preceding commits change when that can happen, but it's a bit tricky to reason about.

How about putting the next commit:

    UploadedAvatarURL [nfc]: Remove `if (match !== null)` check.
    
    `.replace` already has the behavior of doing nothing if a match
    isn't found.

at the beginning? That will fix this bug from the start, and simplify reasoning about the rest.

Reply:
Sure; done. There was a bit of a wrinkle and I put the remove-`g` commit before that one, if that's OK.